src/functions/__tests__/casing.test.ts

```diff
@@ -200,22 +206,124 @@ describe('casing', () => {
       const validWithDigits = ['FOO_BAR1', 'FOO2_4BAR1', 'FOO2_4_2'];
 
       test.each(invalid)('should recognize invalid target %s', target => {
-        expect(runCasing(target, 'macro')).toEqual([{ message: 'must be macro case' }]);
+        expect(runCasing(target, CasingType.macro)).toEqual([{ message: 'must be macro case' }]);
       });
 
       test.each([...valid, ...validWithDigits])('should recognize valid target %s', target => {
-        expect(runCasing(target, 'macro')).toBeUndefined();
+        expect(runCasing(target, CasingType.macro)).toBeUndefined();
       });
 
       describe('when digits are disallowed', () => {
         test.each([...invalid, ...validWithDigits])('should recognize invalid target %s', target => {
-          expect(runCasing(target, 'macro', true)).toEqual([{ message: 'must be macro case' }]);
+          expect(runCasing(target, CasingType.macro, true)).toEqual([{ message: 'must be macro case' }]);
         });
 
         test.each(valid)('should recognize valid target %s', target => {
-          expect(runCasing(target, 'macro', true)).toBeUndefined();
+          expect(runCasing(target, CasingType.macro, true)).toBeUndefined();
         });
       });
     });
   });
+
+  describe('casing with custom separator configuration', () => {
+    describe('parameters validation', () => {
+      const casingTypes = Object.keys(CasingType);
+
+      test.each([...casingTypes])(
+        'with "%s" type - throws when allowLeadingSeparator option is used without specifying separator',
+        type => {
+          expect(() => runCasing('irrelevant value', CasingType[type], undefined, undefined, true)).toThrow(
+            AssertionError,
+          );
+        },
+      );
+
+      const invalidSepLengthCombinations: Array<[string, number]> = [];
+      Object.values(casingTypes).forEach(t =>
```

Comment:
Something to consider:
```suggestion
      const invalidSepLengthCombinations = Object.values(casingTypes).reduce<Array<[string, number]>>((combinations, type) => {
              for (const len of [0, 2, 3, 17]) {
                combinations.push([type, len]);
              }
              
              return combinations;
     }, []);
```

Reply:
Great!